src/include/misc.h

```diff
@@ -86,6 +86,9 @@
 #define WT_MIN(a, b) ((a) < (b) ? (a) : (b))
 #define WT_MAX(a, b) ((a) < (b) ? (b) : (a))
 
+/* Ceil for unsigned integers. */
+#define WT_CEIL(a) ((a) - (uintmax_t)(a) > 0 ? (uintmax_t)(a) + 1 : (uintmax_t)(a))
```

Comment:
Created a custom `WT_CEIL` as it was getting complicated linking the math library to wiredtiger.

Reply:
What was the linking issue? I don't think linker issues are a particularly good rationale for writing our own versions of things. (And while I'm pretty sure your version is correct, I'd still trust the system's libc version over anything custom.)

Reply:
Essentially for optimized builds math library has to be linked to wiredtiger library. This requires some extra work in CMake, and I was having a hard time with it. I saw that we have our custom min and max functions so I decided to do this instead.

Reply:
After asking around and thinking about this a bit more - I think it's right to do our own `ceil`. There isn't that much in this header that's useful for the rest of WT, so the work of fixing the linking issues doesn't get us much.

Reply:
Hi @Siddhartha8899 and @korteland  I don't understand what this function represents: `ceil` is defined to return the integer component of a real number. As I read it: this function increments a value by saying if its unsigned representation is not 0 increment by one.

Reply:
The current macro calculates the difference between a real (double) number and its unsigned representation. If this difference is greater than 0, the macro increments the unsigned representation by one.

Reply:
Sorry I should have though through that more. I realize you're using the float -> int to as floor. However if I'm thinking right this will make `ceil(1.0) = 2` but it should be `ceil(1.0) = 1`.

Reply:
In the scenario where x equals 1.0, the outcome of 1 minus 1 is not greater than 0. As a result, the value of 1 remains unchanged

Reply:
Yes my mistake!

Reply:
All good, thank you for taking the time to review it, @marcbutler.

Reply:
Thanks @Siddhartha8899 I should have taken more time! :)

Reply:
Hi @Siddhartha8899 following up from our conversation.

As `ceil` is defined for negative values, and this macro is not a slightly different name seems apropriate.

Secondly as you have also discussed with @ajmorton there is the problem of finite precision. So as discussed recommend tests using constants from `<float.h>`. You'll find reference here https://en.cppreference.com/w/c/types/limits constants to look at are things `DBL_MIN`, `DBL_TRUE_MIN`, `DBL_EPSILON` and/or the `FLT_*` counterparts.

Reply:
Thanks, @marcbutler updated your first suggestion in [6523ca1](https://github.com/wiredtiger/wiredtiger/pull/9507/commits/6523ca1d24923f85f70429d8be87270987eae615). 

I will create a separate ticket for the precision aspect.

Reply:
Created [WT-11503](https://jira.mongodb.org/browse/WT-11503)